test/e2e/cluster/delete.go

```diff
@@ -48,20 +66,224 @@ func NewDelete(aws aws.Config, logger logr.Logger, endpoint string) Delete {
 	}
 }
 
-func (c *Delete) Run(ctx context.Context, cluster DeleteInput) error {
-	if err := c.deleteCluster(ctx, cluster); err != nil {
-		return fmt.Errorf("deleting EKS hybrid cluster: %w", err)
+func (c *Delete) Run(ctx context.Context, deleteInput DeleteInput) error {
+	filterInput := cleanup.FilterInput{
+		ClusterName:          deleteInput.ClusterName,
+		ClusterNamePrefix:    deleteInput.ClusterNamePrefix,
+		AllClusters:          deleteInput.AllClusters,
+		InstanceAgeThreshold: deleteInput.InstanceAgeThreshold,
+		DryRun:               deleteInput.DryRun,
 	}
 
-	if err := c.stack.delete(ctx, cluster.ClusterName); err != nil {
-		return fmt.Errorf("deleting E2E setup stack: %w", err)
+	if err := c.cleanupEC2Instances(ctx, filterInput); err != nil {
+		return fmt.Errorf("cleaning up EC2 instances: %w", err)
 	}
 
+	if err := c.cleanupIAMRoles(ctx, filterInput); err != nil {
+		return fmt.Errorf("cleaning up IAM roles: %w", err)
+	}
+
+	if err := c.cleanupIAMInstanceProfiles(ctx, filterInput); err != nil {
+		return fmt.Errorf("cleaning up IAM instance profiles: %w", err)
+	}
+
+	if err := c.cleanupCredentialStacks(ctx, filterInput); err != nil {
+		return fmt.Errorf("cleaning up credential stacks: %w", err)
+	}
+
+	if err := c.cleanupEKSClusters(ctx, filterInput); err != nil {
+		return fmt.Errorf("cleaning up EKS clusters: %w", err)
+	}
+
+	if err := c.cleanupArchitectureStacks(ctx, filterInput); err != nil {
```

Comment:
i think for the credentials stack we already have a fail-safe cleanup of individual resources. should we also have it for this one? like vpc, subnets, route tables, vpc peering...